datajoint/heading.py

```diff
@@ -225,6 +237,36 @@ def init_from_database(self, conn, database, table_name):
                     t = re.sub(r' unsigned$', '', t)   # remove unsigned
                     assert (t, is_unsigned) in numeric_types, 'dtype not found for type %s' % t
                     attr['dtype'] = numeric_types[(t, is_unsigned)]
+        
+        #Read and tabulate foreign keys
+        f_keys_query = "SHOW CREATE TABLE `{database}`.`{table_name}`;".format(database = database, table_name=table_name)
```

Comment:
rather than parsing the `CREATE TABLE` statement, please find a way to abstract and re-use the dependency loading functionality in `dependency.py`, starting in line 46.

Reply:
Better yet, simply lookup the dependency in the `dependency` object, if possible.